servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/H2PriorKnowledgeFeatureParityTest.java

```diff
@@ -296,16 +317,26 @@ private void multipleRequests(boolean get, int numberRequests) throws Exception
         }
     }
 
-    @ParameterizedTest(name = "{displayName} [{index}] client={0}, h2PriorKnowledge={1}")
-    @MethodSource("clientExecutors")
-    void cookiesRoundTrip(HttpTestExecutionStrategy strategy, boolean h2PriorKnowledge) throws Exception {
+    @ParameterizedTest(
+            name = "{displayName} [{index}] client={0}, h2PriorKnowledge={1}, strictRfc6265={2}, endsWithSemi={3}")
+    @MethodSource("clientExecutorsCookies")
+    void cookiesRoundTrip(HttpTestExecutionStrategy strategy, boolean h2PriorKnowledge, boolean strictRfc6265,
+                          boolean endsWithSemi)
+            throws Exception {
         setUp(strategy, h2PriorKnowledge);
         InetSocketAddress serverAddress = bindHttpEchoServer();
+        final boolean beforeCookieParsingStrictRfc6265 = H2ToStH1Utils.cookieParsingStrictRfc6265();
+        H2ToStH1Utils.cookieParsingStrictRfc6265(strictRfc6265);
         try (BlockingHttpClient client = forSingleAddress(HostAndPort.of(serverAddress))
                 .protocols(h2PriorKnowledge ? h2Default() : h1Default())
                 .executionStrategy(clientExecutionStrategy).buildBlocking()) {
             HttpRequest request = client.get("/");
-            String requestCookie = "name1=value1; name2=value2; name3=value3";
+            String requestCookie = strictRfc6265 ?
+                    "name1=value1; name2=value2; name3=value3" :
+                    "name1=value1;name2=value2;name3=value3";
```

Comment:
wdyt, should we test `h2PriorKnowledge` scenarios with `DefaultHttpHeadersFactory` too in addition to the `H2HeadersFactory`?

Reply:
I will add another variable to the test config to swap header factories.